### PR TITLE
Chapter1/homework/4

### PR DESCRIPTION
**Pull request by HellHunterMax**

### CODE REVIEW

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 1]
```diff
@@ -0,0 +1,111 @@
+﻿using System;
+using System.Collections.Generic;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Codacy Issue found: [Remove this unnecessary 'using'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4871830)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson4.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4871830&bid=16150848)

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+using System.Text;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Codacy Issue found: [Remove this unnecessary 'using'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4871830)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 4
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson4.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4871830&bid=16150848)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 4
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson4.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4871830&bid=16150848)

**Comment by HellHunterMax** [comment]
Ok ill work on these test results but they are harder than the original Homework. and the Homework did not say we had to account for these problems.

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 4
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson4.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4871830&bid=16150848)

**Comment by HellHunterMax** [comment]
are the tests broken? did i break something? whats wrong?

**Comment by simonasB** [comment]
If you need any help with tests, reply to this comment

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 4
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson4.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4871830&bid=16150848)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 4
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson4.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4871830&bid=16150848)

**Comment by ellmaster** [comment]
I think you are on a good track. I mainly spot newline-errors.

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 4
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson4.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4871830&bid=16150848)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 4
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson4.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4871830&bid=16150848)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 4
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson4.cs  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4871830&bid=16150848)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 4
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson4.cs  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4871830&bid=16150848)

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+            if (name.Equals("-"))
+            {
+                Console.WriteLine("Name cannot be empty.");
+            }
+
+            return name;
+        }
+        public static string PromptString(string message)
+        {
+            Console.Write(message);
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Lesson4.cs]: I think this has to be WriteLine instead.

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+            if (String.IsNullOrEmpty(input))
+            {
+                return "-";
+            }
+
+            return input;
+        }
+        public static int PromptInt(string message)
+        {
+            Console.Write(message);
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Lesson4.cs]: I think this has to be WriteLine instead.

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+            {
+                Console.Write($"{Environment.NewLine}\"{input}\" is not a valid number.");
+                return -1;
+            }
+
+            return number;
+        }
+        public static float PromptFloat(string message)
+        {
+            Console.Write(message);
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Lesson4.cs]: I think this has to be WriteLine instead.

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+
+namespace BootCamp.Chapter
+{
+    class Lesson4
+    {
+        public static void Demo()
+        {
+            PromptPersonData();
+            PromptPersonData();
+        }
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Separate functions with a blank line.

File: Src/BootCamp.Chapter/Lesson4.cs (hunk 5 above)
Comment (Almantask) [reply Src/BootCamp.Chapter/Lesson4.cs]: It's okay, spec was poor.

File: Src/BootCamp.Chapter/Lesson4.cs (hunk 4 above)
Comment (Almantask) [reply Src/BootCamp.Chapter/Lesson4.cs]: It's okay, spec was poor.

File: Src/BootCamp.Chapter/Lesson4.cs (hunk 3 above)
Comment (Almantask) [reply Src/BootCamp.Chapter/Lesson4.cs]: It's okay, spec was poor.

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+                if (isHeightLessThanOrEqualTo0)
+                {
+                    Console.WriteLine($"Height cannot be equal or less than zero, but was {height}.");
+                }
+                return -1f;
+            }
+
+            return weight / (height * height); // BMI = kg/m2
+
+
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Random amount of empty lines

**Pull request closed without merging**